pkg/rancher-desktop/assets/scripts/nerdctl

```diff
@@ -3,4 +3,16 @@ export CONTAINERD_ADDRESS=/run/k3s/containerd/containerd.sock
 if [ -f /usr/local/openresty/nginx/conf/allowed-images.conf ]; then
   export HTTPS_PROXY=http://127.0.0.1:3128
 fi
+
+# On WSL, we need to enter the correct pid &c. namespace for nerdctl to work
+# correctly.
+
+if [ -r /run/wsl-init.pid ]; then
+  parent="$(cat /run/wsl-init.pid)"
+  pid="$(ps -o pid,ppid,comm | awk '$2 == "'"${parent}"'" && $3 == "init" { print $1 }')"
+  if [ -n "${pid}" ]; then
+    exec /usr/bin/nsenter -p -m -n -t "${pid}" /usr/local/libexec/nerdctl/nerdctl "$@"
+  fi
```

Comment:
Should this code be replaced with `wsl-exec`?

```suggestion
  /usr/local/bin/wsl-exec /usr/local/libexec/nerdctl/nerdctl "$@"
```

That way we would inherit any changes to the `nsenter` arguments automatically.

We can do this in a future PR, just so we can get this PR merged today. Maybe add a comment to #4115 to  update the nerdctl wrapper as well?

Reply:
Looking at the `wsl-exec` source I notice that it doesn't get the parent pid, but uses `wsl-init.pid` directly. Why is this needed here, but not in `wsl-exec`?

Reply:
This didn't use `wsl-exec` because it didn't exist two months ago 😝 
As for the parent pid stuff, that's because of changes from the gvisor networking stuff (… same reason as above).  We should just have a follow-up to make this use `wsl-exec` instead and we can ignore that.

Reply:
> As for the parent pid stuff, that's because of changes from the gvisor networking stuff 

But remember, wsl-exec still doesn't lookup parent pid for the default namespace, so why did we ever needed to lookup the parent pid (`/sbin/init`) and not used the child pid (`unshare`), wouldn't that be just adequate?

Reply:
You've got the relationship backwards — the parent is `unshare` (which inherits the outer `wsl-init` pid due to `exec`, and hence is the pid in the pid file), and the child is the `/sbin/init`. The script was looking for the process whose `ppid` is in the pid file (i.e. `/sbin/init`).  The new stuff switches to looking at `unshare` instead. This worked because `unshare` works by putting _itself_ into the requested namespaces, then spawning a child (unless `--no-fork`) which inherits its namespace.

Reply:
> You've got the relationship backwards

Yes, what I meant is this:
```
 # parent="$(cat /run/wsl-init.pid)"
 # ps aux | grep $parent
  421 root      0:00 /usr/bin/unshare --pid --mount-proc --fork --propagation slave /usr/local/bin/wsl-init
 6594 root      0:00 /usr/bin/nsenter -n -p -m -t 421 /bin/sh
```
```
# pid="$(ps -o pid,ppid,comm | awk '$2 == "'"${parent}"'" && $3 == "init" { print $1 }')"
# ps aux | grep $pid
  423 root      0:00 /sbin/init
```

> This worked because unshare works by putting itself into the requested namespaces, then spawning a child (unless --no-fork) which inherits its namespace.

Correct, since we observed that when we were debugging `network-setup` process and that was my point too.